helm/karpenter-bundle/templates/apps.yaml

```diff
@@ -81,8 +81,11 @@ spec:
     inCluster: true
   name: karpenter-crossplane-resources
   namespace: {{ $.Release.Namespace }}
-  version: 0.1.1
+  version: 0.1.2
   extraConfigs:
   - kind: configMap
     name: {{ $.Values.clusterID }}-crossplane-config
     namespace: {{ $.Release.Namespace }}
+  - kind: configMap
```

Comment:
I have some doubts about this one. The requirement is that we need to pass the value `internal.migration.irsaAdditionalDomain` into the `karpenter-crossplane-resources` App (see [here](https://github.com/giantswarm/karpenter-crossplane-resources/pull/22/files#diff-5d59afc6376c8b881d8b0436d86b7b2a79f8a0672190b700eb5b83ac9cce8531R31)), and I've found it available in the `<cluster>-cluster-aws-cluster-values` ConfigMap, which apparently is created by the migration cli, so we could potentially rely on it being present for migrated clusters. But won't be necessarily present on native CAPA clusters. What will happen on those situations? Will the app operator just ignore it or will fail altogether?

Reply:
I tried adding a non existing configmap to the list of `extraConfigs`, and app-operator logged didn't fail, but logged these

```
D 09/02 06:55:56 org-giantswarm/grizzly-external-dns configmap.GetDesiredState checking next layer of extra configs... | app/v7/pkg/values/values.go:122 | controller=app-operator-app | event=update | loop=917778 | version=628636069
D 09/02 06:55:56 org-giantswarm/grizzly-external-dns configmap.GetDesiredState looking for configmap `non-existing-config` in namespace `org-giantswarm` | app/v7/pkg/values/configmap.go:113 | controller=app-operator-app | event=update | loop=917778 | version=628636069
W 09/02 06:55:56 org-giantswarm/grizzly-external-dns configmap.GetDesiredState dependent configMaps are not found | app-operator/v6/service/controller/app/resource/configmap/desired.go:59 | controller=app-operator-app | event=update | loop=917778 | version=628636069
D 09/02 06:55:56 org-giantswarm/grizzly-external-dns configmap.GetDesiredState canceling resource | micrologger@v1.0.0/logger.go:66 | controller=app-operator-app | event=update | loop=917778 | version=628636069
```

which comes from here https://github.com/giantswarm/app-operator/blob/main/service/controller/app/resource/configmap/desired.go

I believe app-operator won't fail, but it won't create the configmap used later by `chart-operator` with the calculated configuration. And we need the configmap or I guess chart-operator won't be able to install the chart. So _I think_ this approach wouldn't work.

Reply:
hmm ok 💩 I'll find another way

Reply:
actually, given that the `<cluster>-userconfig` ConfigMap also contains the necessary property, and that ConfigMap is normally consistently present in both native CAPA and migrated clusters, I'd say it's a safe bet to use that one. What do you think?

Would be nice if we could mark an `extraConfigs` item as optional though.

Reply:
I'm afraid customers could name that configmap differently? But I guess the chances of that happening are very slim, so it may be fine.

Reply:
yeah true, I don't like that either. What if I take `<cluster>-chart-values` from the `giantswarm` namespace. Those are the already merged values passed on to the cluster chart, and that ConfigMap is always created by the app operator. Would that be acceptable? I'm wondering if we already do something like that with another app.

Reply:
After discussing it in standup, we'll go with the `<cluster>-chart-values` CM for this. I've already changed it. Can you review again?